extensions/sql-migration/src/service/serviceClient.ts

```diff
@@ -0,0 +1,255 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SqlOpsDataClient, ClientOptions } from 'dataprotocol-client';
+import { ServerProvider, Events, LogLevel, IConfig } from '@microsoft/ads-service-downloader';
+import { ServerOptions, TransportKind } from 'vscode-languageclient';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+import * as path from 'path';
+import { EventAndListener } from 'eventemitter2';
+import { SqlMigrationService, TdeMigrationService } from './features';
+import { promises as fs } from 'fs';
+import * as constants from '../constants/strings';
+import { IMessage } from './contracts';
+import { ErrorAction, CloseAction } from 'vscode-languageclient';
+import { env } from 'process';
+import { exists } from './utils';
+
+
+export class ServiceClient {
+	private statusView: vscode.StatusBarItem;
+
+	constructor(
+		private outputChannel: vscode.OutputChannel,
+	) {
+		this.statusView = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+	}
+
+	public async startService(context: vscode.ExtensionContext): Promise<SqlOpsDataClient | undefined> {
+		const rawConfig = await fs.readFile(path.join(context.extensionPath, 'config.json'));
+		let clientOptions: ClientOptions = this.createClientOptions();
+		try {
+			let client: SqlOpsDataClient;
+			let serviceBinary: string = '';
+			let downloadBinary = true;
+			if (env['ADS_MIGRATIONSERVICE']) {
```

Comment:
Glad we're adding this, but would suggest documenting it in a README or something so people working on the extension know it exists and how to use it.

Reply:
For the rest of the folks, this env allows us to point to local migration service if we want. To test a locally compiled migration service, you can just set that env before launching ADS. It will pick up the migration service pointed by the directory. So, you don't have to manually copy and paste files every time. 

Reply:
Thanks for clarifying - so on that note, is the development pattern very similar to the existing workflow? e.g., build the migration service locally using dotnet publish, then set an environment variable (in this case a different one than before) to point to where it's built? So basically the same as the existing steps as outlined in https://github.com/microsoft/azuredatastudio/wiki/SQL-Tools-Service, but migration service specific instead of STS as a whole? Or are there any additional steps to be aware of now?

@AkshayMata @aasimkhan30

Reply:
Correct, it's the same general process with just a different env var/path